ggml-quants.c

```diff
@@ -52,6 +52,11 @@
 #define UNUSED GGML_UNUSED
 
 #define MM256_SET_M128I(a, b) _mm256_insertf128_si256(_mm256_castsi128_si256(b), (a), 1)
+#ifdef __GNUC__ 
+#if __GNUC__ < 8
+#define _mm256_set_m128i(a, b) _mm256_permute2f128_si256(_mm256_castsi128_si256(a), _mm256_castsi128_si256(b), 2)
+#endif
+#endif
```

Comment:
clang only claims to be GCC 4.2 according to \_\_GNUC\_\_ and \_\_GNUC\_MINOR\_\_, so this should be:
```suggestion
#if defined(__GNUC__) && __GNUC__ < 8 && !defined(__clang__)
#define _mm256_set_m128i(a, b) _mm256_permute2f128_si256(_mm256_castsi128_si256(a), _mm256_castsi128_si256(b), 2)
#endif
```

Reply:
You are absolutely right, thank you for your patience. Done!

Reply:
> But these calls to _mm256_set_m128i were only added in #&#8203;5187, previously only MM256_SET_M128I was used - you should try removing this macro and using that one instead.

cc @ikawrakow